file/builder.go

```diff
@@ -551,6 +551,9 @@ func (b *stateBuilder) services() {
 }
 
 func (b *stateBuilder) ingestService(s *FService) error {
+	if err := b.addEntityDefaults("services", &s.Service); err != nil {
+		return fmt.Errorf("add defaults to service '%v': %v", *s.Name, err)
+	}
```

Comment:
See line below:
> b.defaulter.MustSet(&s.Service)

Can we please keep the defaulting logic in the defaulter?
Take "Defaulter" as an input to StateBuilder if that is required.